src/kubernetes.cc

```diff
@@ -961,6 +961,38 @@ json::value KubernetesReader::FindTopLevelOwner(
   return FindTopLevelOwner(ns, GetOwner(ns, ref->As<json::Object>()));
 }
 
+bool KubernetesReader::ValidatePodConnectivity() const {
+  try {
+    QueryMaster(std::string(kKubernetesEndpointPath) + "/pods?limit=1");
+
+    return true;
+  } catch (const QueryException& e) {
+    // Already logged.
+    return false;
+  }
+}
+
+bool KubernetesReader::ValidateNodeConnectivity() const {
+  const std::string node_name = CurrentNode();
+  if (node_name.empty()) {
+    return false;
+  }
+
+  try {
+    json::value raw_node = QueryMaster(
+        std::string(kKubernetesEndpointPath) + "/nodes/" + node_name);
+
+    return true;
+  } catch (const QueryException& e) {
+    // Already logged.
+    return false;
+  }
+}
+
+bool KubernetesReader::ValidateConfiguration() const {
+  return ValidatePodConnectivity() && ValidateNodeConnectivity();
```

Comment:
We might also fail if both `KubernetesUseWatch` and `KubernetesUpdaterIntervalSeconds` are enabled.

Reply:
I agree and disagree with this. Being that the other polling updaters are not considered valid / invalid based on the polling interval, I think we should consolidate this to whether or not the pollers are able to communicate with endpoints, not whether or not you have a poll interval > 0.

Reply:
I think this depends on whether you want to get an error message when poll interval < 0 or just silently avoid polling. Personally, I prefer the former (and reserve poll interval == 0 for "do not poll").

Perhaps we should implement `ValidateConfiguration` in `PollingMetadataUpdater` to verify the polling interval for all polling updaters, and invoke `PollingMetadataUpdater::ValidateConfiguraiton` here before doing updater-specific validation?..

Reply:
Totally on board with this, updated.